ComputePipeline/src/ComputePipeline.jl

```diff
@@ -74,7 +74,8 @@ end
 A `ComputeEdge` represents the computation connecting one set of nodes to another.
 They are considered internal and should not be interacted with or created directly.
 """
-struct ComputeEdge <: AbstractEdge
+struct ComputeEdge{T} <: AbstractEdge
+    graph::T
```

Comment:
I wonder if it's better to do something like 
```julia
abstract type AbstractComputeGraph end

struct ComputeEdge <: AbstractEdge
    graph::AbstractComputeGraph
    ...
end

struct ComputeGraph <: AbstractComputeGraph ... end
```
for these. Iirc we had a case where this was better for performance already, maybe with ComputeEdge?

Reply:
But we only have one type here - I think the performance improvement came from it having several types in one array, making iteration etc slow.

Reply:
Alright, I checked, it's significantly worse to do what I proposed.